### PR TITLE
Add method to obtain server in TickEvent.ServerTickEvent

### LINKED ISSUE #2724: ServerTickEvent doesn't support multiple servers
In 1.9 it appears Mojang are preparing to have multiple MinecraftServer instances in the same process.

ServerTickEvent doesn't tell you which server is ticking. It just needs a MinecraftServer field.

For consistency, maybe ClientTickEvent should have a Minecraft field, but the client is still a singleton anyway.

### PR DESCRIPTION
**Pull request by kashike**
Fixes #2724

### CODE REVIEW

File: src/main/java/net/minecraftforge/fml/common/FMLCommonHandler.java [hunk 1]
```diff
@@ -248,9 +248,18 @@ public IFMLSidedHandler getSidedDelegate()
...
     }
 
+    /**
+     * @deprecated use {@link #onPostServerTick(MinecraftServer)}
+     */
+    @Deprecated
     public void onPostServerTick()
     {
-        bus().post(new TickEvent.ServerTickEvent(Phase.END));
+        this.onPreServerTick(this.getMinecraftServerInstance());
```
Comment (liach) [inline src/main/java/net/minecraftforge/fml/common/FMLCommonHandler.java]:
```java
this.onPostServerTick(this.getMinecraftServerInstance());
```

Comment (kashike) [inline src/main/java/net/minecraftforge/fml/common/FMLCommonHandler.java]: I've fixed this locally, will push in a few minutes once `genPatches` finishes - thanks

**Comment by kashike** [comment]
Is there a reason for this not being merged?

**Comment by kashike** [comment]
I don't see an obvious reason for why this was closed - could you enlighten me?

**Pull request closed without merging**